### PR TITLE
process-cpp: init at unstable-2021-05-11

### PR DESCRIPTION
**Pull request by onny**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/process-cpp/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    boost
+    properties-cpp
+  ];
+
+  meta = with lib; {
+    description = "A simple convenience library for handling processes in C++11";
+    homepage = "https://gitlab.com/ubports/development/core/lib-cpp/process-cpp";
+    license = with licenses; [ gpl3Only lgpl3Only ];
+    maintainers = with maintainers; [ onny ];
+  };
```
Comment (OPNA2608) [inline pkgs/development/libraries/process-cpp/default.nix:42]: Fails to build on Darwin CI.

```
       > [ 11%] Building CXX object src/CMakeFiles/process-cpp.dir/core/posix/child_process.cpp.o
       > /tmp/nix-build-process-cpp-unstable-2021-05-11.drv-0/source/src/core/posix/child_process.cpp:33:10: fatal error: 'sys/eventfd.h' file not found
       > #include <sys/eventfd.h>
       >          ^~~~~~~~~~~~~~~
       > 1 error generated.
```

AFAICT this header seems Linux-specific? Seems normally unavailable on Darwin at least.

```suggestion
    platforms = platforms.linux;
  };
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]

**Comment by onny** [comment]
Thank you for this really helpful review :) Added the fixes

**Review by OPNA2608 — APPROVED** [review]
Result of `nixpkgs-review pr 207839` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>process-cpp</li>
  </ul>
</details>

---

Result of `nixpkgs-review pr 207839` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>process-cpp</li>
  </ul>
</details>

---

No downstream consumer of this in Nixpkgs yet to verify its functionality but everything builds fine. There are some Doxygen docs that *could* be built & subsequently moved to a `doc` output, but as-is this LGTM.

**Comment by onny** [comment]
Thank you for reviewing :)

**Pull request merged**